src/hotspot/share/gc/shenandoah/shenandoahGenerationalHeap.cpp

```diff
@@ -330,11 +315,11 @@ oop ShenandoahGenerationalHeap::try_evacuate_object(oop p, Thread* thread, Shena
   }
 
   // Try to install the new forwarding pointer.
-  ContinuationGCSupport::relativize_stack_chunk(copy_val);
-
   oop result = ShenandoahForwarding::try_update_forwardee(p, copy_val);
   if (result == copy_val) {
     // Successfully evacuated. Our copy is now the public one!
+    ContinuationGCSupport::relativize_stack_chunk(copy_val);
```

Comment:
I don't remember the significance of relative_stack_chunk().  Should we have a comment to explain what this is doing?

Reply:
I don't really know what `relativize_stack_chunk` is doing either (something to do with Virtual Threads). The important thing here is that it may inspect the header without realizing the header is a forwarding pointer. Secondarily, we don't want to do work on oops that aren't going to be visible (i.e, it lost the evacuation race). I'll add a comment here.